rotkehlchen/db/addressbook.py

```diff
@@ -0,0 +1,108 @@
+import sqlite3
+from typing import List, Optional
+
+from eth_utils import to_checksum_address
+from pysqlcipher3 import dbapi2
+
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.errors.misc import InputError
+from rotkehlchen.globaldb import GlobalDBHandler
+from rotkehlchen.types import AddressbookEntry, AddressbookType, ChecksumEthAddress
+
+
+class DBAddressbook:
+
+    def __init__(self, db_handler: DBHandler) -> None:
+        self.db = db_handler
+
+    def _get_connection(self, book_type: AddressbookType) -> sqlite3.Connection:
+        if book_type == AddressbookType.GLOBAL:
+            return GlobalDBHandler().conn
+        return self.db.conn
+
+    def get_addressbook_entries(
+        self,
+        book_type: AddressbookType,
+        addresses: Optional[List[ChecksumEthAddress]] = None,
+    ) -> List[AddressbookEntry]:
+        connection = self._get_connection(book_type)
+        cursor = connection.cursor()
+        if addresses is None:
+            cursor.execute('SELECT address, name FROM address_book')
+        else:
+            questionmarks = ','.join('?' * len(addresses))
+            cursor.execute(
+                f'SELECT address, name FROM address_book WHERE address IN ({questionmarks})',
+                addresses,
+            )
+
+        entries = []
+        for address, name in cursor:
+            entries.append(AddressbookEntry(address=to_checksum_address(address), name=name))
+
+        return entries
+
+    def add_addressbook_entries(
+        self,
+        book_type: AddressbookType,
+        entries: List[AddressbookEntry],
+    ) -> None:
+        connection = self._get_connection(book_type)
+        cursor = connection.cursor()
+        # We iterate here with for loop instead of executemany in order to catch
+        # which identifier is duplicated
+        # We raise an error here in order to help debugging. The errors should not appear in
+        # production if the frontend handles the logic properly
```

Comment:
This comment is not needed. The api is designed (also the doc description) to raise an error here.

```suggestion
```